src/utils/url/Url.ts

```diff
@@ -1,12 +1,25 @@
 import { IUrl } from "./IUrl";
+import { Protocol } from "./Protocol";
+import { copyMutateAndWrap } from "./copyMutateAndWrap";
+import { defaultPort } from "./defaultPort";
 
-export const Url = (urlString:string):IUrl => ({
-    get path() { return new URL(urlString).pathname },
-    asString: urlString,
-    setPath(newPath) { 
-        const tempUrl = new URL(urlString);
-        tempUrl.pathname = newPath
-        return Url(tempUrl.toString())
-    },
-    toString: () => urlString,
+export const Url = (nativeURL:URL):IUrl => ({
+    get path() { return nativeURL.pathname },
+    get host() { return nativeURL.host },
```

Comment:
would `fullyQualifiedDomainName` be a better name for us to switch to?

Reply:
I think so!